blocks/api/raw-handling/shortcode-converter.js

```diff
@@ -32,11 +32,22 @@ function segmentHTMLToShortcodeBlock( HTML ) {
 	const transformTag = first( transformTags );
 
 	let match;
-	let lastIndex = 0;
 
 	if ( ( match = shortcode.next( transformTag, HTML, lastIndex ) ) ) {
+		const beforeHTML = HTML.substr( 0, match.index );
+
 		lastIndex = match.index + match.content.length;
 
+		// If the shortcode content does not contain HTML and the shortcode is
+		// not on new line (or in paragraph from Markdown converter), consider
+		// the shortcode as inline text.
```

Comment:
I'd add at the end of the sentence: _, and thus skip conversion for this segment._ Does it make sense?